src/server/wasm/wasm_registry.cc

```diff
@@ -0,0 +1,135 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "server/wasm/wasm_registry.h"
+
+#include <server/wasm/api.h>
+
+#include <shared_mutex>
+
+#include "absl/cleanup/cleanup.h"
+#include "absl/strings/str_cat.h"
+#include "base/logging.h"
+#include "io/file_util.h"
+#include "server/wasm/api.h"
+#include "wasm.h"
+#include "wasmtime.h"
+
+namespace dfly::wasm {
+
+WasmRegistry::WasmRegistry() {
+  engine_ = wasm_engine_new();
+  CHECK(engine_);
+  linker_ = wasmtime_linker_new(engine_);
+  CHECK(linker_);
+  CHECK_EQ(wasmtime_linker_define_wasi(linker_), nullptr);
+
+  // There are more configs we should look at. In particular we can spend a little more
+  // time in compiling but generate a more optimized binary as well ass signal handling config.
+  wasi_config_ = wasi_config_new();
+  CHECK(wasi_config_);
+  wasi_config_inherit_argv(wasi_config_);
+  wasi_config_inherit_env(wasi_config_);
+  wasi_config_inherit_stdin(wasi_config_);
+  wasi_config_inherit_stdout(wasi_config_);
+  wasi_config_inherit_stderr(wasi_config_);
```

Comment:
is there a way to start it without wasi? We don't need it, it should be a pure sandbox

Reply:
`wasi` is the portable abstraction of a system interface (for example reading from a socket `read`). Some of these are implemented while others have limited functionality. Either way, I think it's fine to have those.